autogen.pl

```diff
@@ -776,6 +776,43 @@ sub find_and_delete {
 # Find a specific executable and ensure that it is a recent enough
 # version.
 
+sub check_m4_gnu_extensions {
+    verbose "Checking whether m4 supports the gnu extensions\n";
+    my $m4 = $ENV{'M4'};
+    $m4 = "m4" unless (defined $m4);
+    my $result = `$m4 -g config/check_m4_gnu_extension.m4`;
+    if ($? != 0) {
+        $result = `$m4 config/check_m4_gnu_extension.m4`;
```

Comment:
My point here was that we should check here if `m4` fails, too.  The first time you call `m4`, you can expect it to fail for the "on the wrong OS" reason.  But `m4` could fail for some unexpected reason, too.  And that `m4` failure output may not go to stdout, making troubleshooting quite difficult.  So it would be good to check here and fail nicely if `m4` fails for an unexpected reason.